core/src/jsMain/kotlin/Options.kt

```diff
@@ -0,0 +1,49 @@
+package com.juul.kable
+
+/** https://developer.mozilla.org/en-US/docs/Web/API/Bluetooth/requestDevice */
+public data class Options(
+    val optionalServices: Array<String> = emptyArray(),
+    val filters: Array<Filter>? = null,
+) {
+
+    public sealed class Filter {
+
+        public data class Name(val name: String) : Filter()
+
+        public data class NamePrefix(val namePrefix: String) : Filter()
+
+        public data class Services(val services: Array<String>) : Filter() {
+            override fun equals(other: Any?): Boolean {
+                if (this === other) return true
+                if (other == null || this::class.js != other::class.js) return false
+                other as Services
+                if (!services.contentEquals(other.services)) return false
+                return true
+            }
+
+            override fun hashCode(): Int {
+                return services.contentHashCode()
+            }
+        }
```

Comment:
Nit: Would it make sense to just use `List` instead of `Array` for these things? Would save a _lot_ of boilerplate.

Reply:
Does that decisions have something to do with JS and how it's handled there?

Reply:
@sdonn3 is correct, the `Array` is a more natural collection for interop with Javascript (`List` works, but uses transpiled `List` class from stdlib rather than native JS types).

Relevant snippets of documentation below:

---

- `optionalServices[]`: An array of `BluetoothServiceUUID`s.

— [Bluetooth.requestDevice()](https://developer.mozilla.org/en-US/docs/Web/API/Bluetooth/requestDevice)

---

- `kotlin.Array` is mapped to JavaScript Array.
- Kotlin collections (`List`, `Set`, `Map`, etc.) are not mapped to any specific JavaScript type.

— [Representing Kotlin types in JavaScript](https://kotlinlang.org/docs/reference/js-to-kotlin-interop.html#representing-kotlin-types-in-javascript)

Reply:
👍  gotcha. A little unfortunate, but since we're doing this in `jsMain` and not `common` I think it's totally fine.